OpenRA.Game/Graphics/HardwarePalette.cs

```diff
@@ -60,6 +61,14 @@ public void AddPalette(string name, ImmutablePalette p, bool allowModifiers)
 			int index = palettes.Count;
 			indices.Add(name, index);
 			palettes.Add(name, p);
+
+			if (palettes.Count > buffer.GetLength(0))
+			{
+				var oldBuffer = buffer;
+				buffer = new uint[Exts.NextPowerOf2(palettes.Count), Palette.Size];
```

Comment:
I've never been a fan of these two dimensional arrays.  Would it be worth changing to a 1D array, and then using `Array.Resize`?